### PR TITLE
feat: several major improvements

### PR DESCRIPTION
**Pull request by dbluhm**
This PR implements a number of improvements.

I spent a fair amount of time just cleaning up styling and layout. It should look a bit better (at least subjectively to me). I've also implemented:

- Connect and disconnect toggle button
- A message card for discover feature messages
- Manual composition, with one sample message for trust ping
- Fixed messages out of order in message history
- Fixed the retrieve messages and connect/disconnect disappearing on smaller screen sizes.

Finally the code was formatted with prettier. This should be happening on every check in but since we've not set up an action or git hooks, I had to do this manually.

### CODE REVIEW

File: src/pages/profile/jsoneditor.ts [hunk 1]
```diff
@@ -1,37 +1,44 @@
-import * as m from "mithril"
-import { CodeJar } from "codejar"
```
Comment (frostyfrog) [inline src/pages/profile/jsoneditor.ts:2]: Is CodeJar used anywhere else? I don't see it being removed from the dependency list, but it was removed from the JsonEditor file.

**Review by frostyfrog — COMMENTED** [review]
In general, the changes look good to me, however, I've left some comments where I have some questions/concerns.

File: src/pages/profile/jsoneditor.ts (hunk 1 above)
Comment (dbluhm) [reply src/pages/profile/jsoneditor.ts:2]: No longer used; I'll push a commit to take care of that!

**Comment by dbluhm** [comment]
For the formatting concerns raised, if you know the right options to set for .prettierrc, I'd be happy to make the formatter handle this for us lol. Otherwise, I'd say let's not hold up this PR for formatting decisions and we can log an issue to address in a follow up.

**Comment by frostyfrog** [comment]
That's fair in terms of formatting. We can look at that later. Did you see my comment about the return type of `onMessage` and `onAnyMessage`?

**Comment by dbluhm** [comment]
Oh yeah, fair recommendation, I'll push a change for that

File: src/lib/peer2.ts [hunk 2]
```diff
@@ -110,23 +114,27 @@ export default class DIDPeer {
...
     const abbreviated = Object.fromEntries(
       Object.entries(input).map(([key, value]) => {
         const abbreviatedKey = commonStringAbbreviations[key] || key
-        const abbreviatedValue = typeof value === 'string'
-          ? (commonStringAbbreviations[value] || value)
-          : value
+        const abbreviatedValue =
+          typeof value === "string"
+            ? commonStringAbbreviations[value] || value
+            : value
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply src/lib/peer2.ts:134]: To be addressed in another PR

File: src/lib/peer2.ts [hunk 3]
```diff
@@ -110,23 +114,27 @@ export default class DIDPeer {
     const expanded = Object.fromEntries(
       Object.entries(input).map(([key, value]) => {
         const expandedKey = reverseCommonStringAbbreviations[key] || key
-        const expandedValue = typeof value === 'string'
-          ? (reverseCommonStringAbbreviations[value] || value)
-          : value
+        const expandedValue =
+          typeof value === "string"
+            ? reverseCommonStringAbbreviations[value] || value
+            : value
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply src/lib/peer2.ts:120]: To be addressed in another PR

File: src/lib/agent.ts [hunk 4]
```diff
@@ -62,99 +74,105 @@ export class Agent {
...
 
   public onMessage(type: string, callback: (message: AgentMessage) => void) {
-    eventbus.on(type, callback)
+    return eventbus.on(type, callback)
   }
 
   public onAnyMessage(callback: (message: AgentMessage) => void) {
-    eventbus.on("messageReceived", callback);
+    return eventbus.on("messageReceived", callback)
   }
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply src/lib/agent.ts]: Fixed!

File: src/lib/agent.ts [hunk 5]
```diff
@@ -62,99 +74,105 @@ export class Agent {
...
-    }
-  }
+    const from =
+      message.from == this.profile.did
+        ? (this.profile as Contact)
+        : ContactService.getContact(message.from)
+    const to =
+      message.to[0] == this.profile.did
+        ? (this.profile as Contact)
+        : ContactService.getContact(message.to[0])
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply src/lib/agent.ts:145]: To be addressed in another PR

File: src/lib/agent.ts [hunk 6]
```diff
@@ -62,99 +74,105 @@ export class Agent {
...
   }
 
   private handleDiscoverFeatures(message: IMessage) {
-    const regexEscape = (s: string) => s.replace(/([.*+?$^=!:{}()|\[\]\/\\])/g, "\\$1");
-    const createRegex = (query: string) => (new RegExp(`^${query.split("*").map(regexEscape).join(".*")}$`));
-    let protocolResponse: object[] = [];
+    const regexEscape = (s: string) =>
+      s.replace(/([.*+?$^=!:{}()|\[\]\/\\])/g, "\\$1")
+    const createRegex = (query: string) =>
+      new RegExp(`^${query.split("*").map(regexEscape).join(".*")}$`)
```
[the remark this answers is not in the document]
Comment (dbluhm) [reply src/lib/agent.ts:91]: To be addressed in another PR

**Review by frostyfrog — APPROVED** [review]
LGTM

**Pull request merged**